### PR TITLE
EIP4844: Add some e2e tests.

### PR DESCRIPTION
**Pull request by EvanJRichard**
**Description**

This seeks to cherry pick some `op-e2e` tests from #7349, as well as adding simple additional tests.

**Tests**
- [ ] simple e2e system test that sends blobs on l1
- [ ] update `fakepos` to use `v3`
- [ ] Pull whatever tests from #7349 that can be easily pulled 

**Metadata**
See also: https://github.com/ethereum-optimism/optimism/pull/7993
See also: https://github.com/ethereum-optimism/optimism/pull/8115

### CODE REVIEW

**Comment by EvanJRichard** [comment]
Looks like I grabbed way too much. Unmangling in process.

File: op-e2e/e2eutils/fakebeacon/blobs.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	return nil
+}
+
+func (f *FakeBeacon) LoadBlobsBundle(slot uint64) (*engine.BlobsBundleV1, error) {
+	f.blobsLock.Lock()
+	defer f.blobsLock.Unlock()
+	bundlePath := fmt.Sprintf("blobs_bundle_%d.json", slot)
+	data, err := os.ReadFile(filepath.Join(f.blobsDir, bundlePath))
+	if err != nil {
+		if os.IsNotExist(err) {
```
[the remark this answers is not in the document]
Comment (EvanJRichard) [reply op-e2e/e2eutils/fakebeacon/blobs.go]: Ah, right, the reset reintroduced this.

**Comment by EvanJRichard** [comment]
Cross reference with https://github.com/ethereum-optimism/optimism/pull/8115

File: op-e2e/eip4844_test.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// TestSystem4844E2E runs the SystemE2E test with 4844 enabled on L1,
+// and active on the rollup in the op-batcher and verifier.
+func TestSystem4844E2E(t *testing.T) {
```
Comment (protolambda) [inline op-e2e/eip4844_test.go:22]: Pulling in the full e2e 4844 test won't work until the 4844 batcher/verifier functionality lands.

File: op-service/eth/types.go [hunk 3]
```diff
@@ -142,8 +144,7 @@ type ExecutionPayload struct {
 	ExtraData     BytesMax32      `json:"extraData"`
 	BaseFeePerGas Uint256Quantity `json:"baseFeePerGas"`
 	BlockHash     common.Hash     `json:"blockHash"`
-	// nil if not present, pre-shanghai
-	Withdrawals *types.Withdrawals `json:"withdrawals,omitempty"`
+	Withdrawals   *Withdrawals    `json:"withdrawals,omitempty"`
```
Comment (protolambda) [inline op-service/eth/types.go:147]: The withdrawals type shouldn't be included; it's a remnant of a previous Canyon iteration that the eip4844 branch is based on, develop is more up-to-date here, and uses the geth typing.

File: op-e2e/actions/l2_engine.go [hunk 4]
```diff
@@ -176,8 +185,22 @@ func (e *L2Engine) ActL2IncludeTx(from common.Address) Action {
...
+		err := wait.For(ctx, time.Second, func() (bool, error) {
+			i = e.engineApi.PendingIndices(from)
+			txs, q = e.eth.TxPool().ContentFrom(from)
+			return uint64(len(txs)) > i, nil
+		})
+		require.NoError(t, err,
+			"no pending txs from %s, and have %d unprocessable queued txs from this account: %w", from, len(q), err)
+
+		tx := txs[i]
+		err = e.engineApi.IncludeTx(tx, from)
```
Comment (protolambda) [inline op-e2e/actions/l2_engine.go:203]: This diff has nothing to do with the 4844 diff, seems like older develop code that is re-introduced? We can drop this.

File: op-e2e/setup.go [hunk 5]
```diff
@@ -273,16 +276,18 @@ func (sys *System) NodeEndpoint(name string) string {
...
 	if sys.BatchSubmitter != nil {
-		_ = sys.BatchSubmitter.Kill()
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		sys.BatchSubmitter.StopIfRunning(ctx)
 	}
 
+	postCtx, postCancel := context.WithCancel(context.Background())
+	postCancel() // immediate shutdown, no allowance for idling
+
```
Comment (protolambda) [inline op-e2e/setup.go:290]: Why change the batch submitter start/stop handling? `develop` is already up to date here, this diff breaks it

**Comment by protolambda** [comment]
Rebased the `eip4844` branch and fixed conflicts with `develop` there, and then cherry-picked the e2e test utils and work from the 4844 branch, and added additional tests. See https://github.com/ethereum-optimism/optimism/pull/8131 Closing this. Thank you for the effort, but with the erroneously re-introduced old `develop` contents I thought it was cleaner to do the rebase + pull in the right contents + write an e2e test that runs L1 Dencun but not the L2 DA changes yet.

**Pull request closed without merging**